ThePalaceProject/circulation-admin
# Upgrade to Node 18.

## Description

Upgrades this project to build with Node 18, along with some other quality/security improvements. This includes:

- Upgrading dependencies that were incompatible with Node 18
- Upgrading dependencies to resolve `npm audit` security errors (not all of them, just the easy ones for now)
- Code/configuration changes required by major version upgrades of dependencies, including
  - Major webpack update required updating some webpack config files and removing some no-longer-supported arguments
- Some minor formatting changes to resolve lint errors
- Additional `export`s from some files to resolve documentation generation warnings

## Motivation and Context

The project had been using Node 12, which is no longer supported, and will not be allowed in GitHub actions soon. Notion: https://www.notion.so/lyrasis/Upgrade-circulation-admin-to-Node-18-f5591621152e44e8865397700f9bfdb4

## How Has This Been Tested?

- Verify that `npm install` runs without error
- Verify that all npm scripts in package.json run without error (except for the nightwatch/selenium tests `npm run test-browser`, `npm run test-chrome`, and `npm run test-firefox`, which already weren't fully working). This especially includes `npm test`.
- Verify that the app runs in dev:
  - `npm run dev-server -- --env=backend=https://cerberus.staging.palaceproject.io`
  - Open http://localhost:8080/admin in a browser
  - Log in using cerberus credentials
  - Verify that all UI functionality works

## Checklist:
